### ISSUE TITLE
Allow passthrough of extra/unknown attributes in MML

### ISSUE THREAD

**Issue by jrunestone** [MyraUI/Myra#241]
Hi! I'm seeing a personal need of extending the MML with custom widget attributes (think attributes for data binding and events). The project imports without errors when I add arbitrary attributes to the MML but as far as I can see the unknown attributes are discarded and not stored on the widgets after importing. Is there a way to access the custom attributes right now or would that be a fitting feature to request or submit a PR for? I'm thinking just a simple dictionary of attribute keys and values on each widget with either all "raw" attributes or just the unparsed attributes.

Thank you.

**Comment by rds1983** [comment]
I think the easiest way to archieve that is to simply add property
```c#
  [DefaultValue(null)]
  public string UserData { get; set; }
```
to the BaseObject class(which is inherited by Widget, MenuItem, TabItem and a few more).
Though if a user would want to store a dictionary of values here, they would need to store values like ("a=5;b=7") and write parsing code themselves. But it's not a big deal.
I would gladly accept such PR.

**Comment by jrunestone** [comment]
I was thinking more like putting each attribute that couldn't be paired to a property (in LoadContext) into a dictionary with its value, and then storing it on the BaseObject. Is there a problem with introducing a dictionary property on the BaseObject?

```
<!-- File.mml -->
<Label MyCustomAttribute="foo" />
```

```
// BaseObject
[XmlIgnore]
public IDictionary<string, string> UserData { get; set; };
```

```
// LoadContext
if (!propertyWasSet)
{
    obj.UserData.Add(attr.Name, attr.Value);
}
```

```
// User code
var customAttrValue = Desktop.Root.Widgets[0].UserData["MyCustomAttribute"]; // foo
```

**Comment by rds1983** [comment]
This approach has a flaw: a custom property name might turn real one in the future version of Myra and the user code would stop working properly after the update.
Probably it makes sense to add additional constraint: a custom properties should start with symbol '_'.
I.e.
```
<Label _MyCustomAttribute="foo" />
```

**Comment by jrunestone** [comment]
Yeah you're right, a prefix might be prudent.

**Comment by jrunestone** [comment]
Resolved in PR #244
